core/src/main/java/com/google/zxing/datamatrix/encoder/EdifactEncoder.java

```diff
@@ -66,6 +66,11 @@ private static void handleEOD(EncoderContext context, CharSequence buffer) {
         context.updateSymbolInfo();
         int available = context.getSymbolInfo().getDataCapacity() - context.getCodewordCount();
         int remaining = context.getRemainingCharacters();
+        // The following two lines are a hack inspired by the 'fix' from https://sourceforge.net/p/barcode4j/svn/221/
+        if (remaining > available) {
+          context.updateSymbolInfo(context.getCodewordCount()+1);
```

Comment:
The `+` needs spaces around it to pass checkstyle, but otherwise seems fine.

Reply:
Sorry, I don't get it. Where are spaces missing?

Reply:
Around the + operator at the end of the line

Reply:
Aargh, you are right. I looked at the plus sign at the beginning of the line.